adapters/flipp/flipp.go

```diff
@@ -219,7 +219,32 @@ func buildBid(decision *InlineModel, impId string) *openrtb2.Bid {
 		if decision.Contents[0].Data.Width != 0 {
 			bid.W = decision.Contents[0].Data.Width
 		}
-		bid.H = 0
+		customDataInterface := decision.Contents[0].Data.CustomData
+		customDataMap, ok := customDataInterface.(map[string]interface{})
+		var defaultStandardHeight int64 = 2400
+		var defaultCompactHeight int64 = 600
+		var flippExtParams openrtb_ext.ImpExtFlipp
+		if flippExtParams.Options.StartCompact {
+			if ok { // check if customDataMap exists and startCompact is true
+				if height, exists := customDataMap["compactHeight"].(int64); exists {
+					bid.H = height
+				} else {
+					bid.H = defaultCompactHeight
+				}
+			} else {
+				bid.H = defaultCompactHeight
+			}
+		} else { // startCompact is false
+			if ok { // customDataMap exists
+				if height, exists := customDataMap["standardHeight"].(int64); exists {
+					bid.H = height
+				} else {
+					bid.H = defaultStandardHeight
+				}
+			} else {
+				bid.H = defaultStandardHeight
+			}
+		}
 	}
```

Comment:
You can simplify this a little further by doing

```go
if flippExtParams.Options.StartCompact {
	defaultHeight = defaultCompactHeight
} else {
	defaultHeight = defaultStandardHeight
}

bid.H = defaultHeight

if ok { 
	if height, exists := customDataMap[flippExtParams.Options.StartCompact ? "compactHeight" : "standardHeight"].(int64); exists {
		bid.H = height
	}
}
```

Reply:
> ```go
> 	if height, exists := customDataMap[flippExtParams.Options.StartCompact ? "compactHeight" : "standardHeight"].(int64); exists {
> ```

I get an error for the ? operator, saying **expected ']', found 'ILLEGAL'syntax**

Reply:
Sorry 🤦‍♂️ you can do this instead

```go
if flippExtParams.Options.StartCompact {
	defaultHeight = 600 // Default compact height
} else {
	defaultHeight = 2400 // Default standard height
}

// Set default height first
bid.H = defaultHeight

if ok { // customDataMap exists
	var key string
	if flippExtParams.Options.StartCompact {
		key = "compactHeight"
	} else {
		key = "standardHeight"
	}

	if height, exists := customDataMap[key].(int64); exists {
		bid.H = height
	}
}
```